src/modules/check/connect.ts

```diff
@@ -29,7 +29,7 @@ function connect(options: WalletCheckCustomOptions = {}): WalletCheckModule {
       }
     }
 
-    if (!stateStore.address.get() && wallet && wallet.name) {
+    if (!stateStore.address.get() && wallet && wallet.name && wallet.name !== 'CoboVault') {
```

Comment:
I would suggest that we take this clause out that checks for CoboVault. I can see why you don't want the connect modal to render as it looks a little weird underneath the qr scanner modal that pops up. I think this modal is important as it gives the user some context, especially if the user clicks cancel on the cobo scanner modal, then the connect modal is there explaining that this step is necessary to continue with the app and allows them to click "Try again" to show the qr modal again.

Maybe you could tweak your modal design a bit so that it is full screen and darkens the background which would block out the connect modal until the user clicks cancel? Check out the WalletConnect modal on our react demo for an example of what that could look like.

Reply:
Hi @aaronbarnardsound, thanks a lot for your feedback, initially I think the Cobo Vault does not have a `connect` step in our user journey, because the authorization is a default behavior when user want to sync his Cobo Vault to a watch-only wallet via QR codes. 
So it will looks a little weird that I sync my wallet QR codes but nothing happened, then I need to sync my QR codes again in another QR scanner modal. I think it is caused by some inconsistent business logic. 
And I think we can revert this change and modify Cobo Vault's `connect` modal by adding some help texts like "Please scan Cobo Vault Sync QR codes for authorizing" and adding other help texts to `accounts` modal like "Please sync Cobo Vault to get accounts" to make it clear.
What do you think? Any better suggestions?

Reply:
@soralit My concern is not as much for showing the modal informing the user to connect, but what happens if the user clicks "close" on your QR modal without scanning their wallet, it gets stuck on this screen:

<img width="241" alt="Screen Shot 2021-03-24 at 10 48 40 am" src="https://user-images.githubusercontent.com/29873495/112233492-a667d780-8c8e-11eb-8e73-e7dc507900d3.png">

Whereas if you include the Onboard connect modal it displays this:

<img width="667" alt="Screen Shot 2021-03-24 at 10 51 35 am" src="https://user-images.githubusercontent.com/29873495/112233694-f8106200-8c8e-11eb-8781-e9e7baa2a659.png">

This modal allows the user to click "Try Again" button which will display the QR modal again. So i think it is best to make the suggested change above.

Reply:
👍